File: ReactNativeApp/ios/Modules/Phases/Games/GameSelector/GameSelector.swift
//
//  GameSelector.swift
//  jogo
//
//  Created by arham on 19/11/2021.
//

import Foundation

public enum GamePhasesName: String {
  case Juggling = "Juggling"
  case GraphGame = "GraphGame"
  case JugglingGraph = "JugglingGraph"
}

class GameSelector {
  
  static func select(name: GamePhasesName) -> GamePhase.Type? {
    switch name {
    case .Juggling:
      return JugglingPhase.self
    case .GraphGame:
      return GraphGamePhase.self
    case .JugglingGraph:
      return GraphGamePhase.self
    }
  }
}

File: ReactNativeApp/android/app/src/main/java/com/phases/games/GameSelector.java
package com.phases.games;

import com.phases.games.instances.JugglingPhase;
import com.phases.instances.GamePhase;

public class GameSelector {

    public static GamePhase select(String gameName) {

        switch (gameName) {
            case "Juggling":
                return new JugglingPhase();
        }
        return null;
    }

    public enum Games {
        Juggling
    }

}
